### PR TITLE
Implement closed_stream event handling and enable related tests

### PR DESCRIPTION
**Pull request by bomanaps**
## What was wrong?

Issue #https://github.com/libp2p/py-libp2p/issues/825

## How was it fixed?
This PR completes the core “closed stream” functionality and ensures tests now validate it.

### Linked discussion  
This issue stems from discussions here: https://github.com/libp2p/py-libp2p/discussions/724

Summary of approach.
### What’s completed in this PR:
- [x] Implement `closed_stream` in `MyNotifee` (#826)
- [x] Implement swarm notification for closed stream (`notify_closed_stream`) (#828)
- [x] Add remove_stream hook in SwarmConn for stream closure integration (#829)
- [x] Enable tests for `closed_stream` in `test_notify.py` and `test_notify_stream_correctness` (#830 partially)

### What remains to be done:
- [ ] Implement `listen_close` in `MyNotifee` (#827)
- [ ] Implement swarm notification for listener closure (`notify_listen_close`) (#828)
- [ ] Add integration points for listener lifecycle (e.g. calling `notify_listen_close`) (#829)
- [ ] Write tests for `listen_close` (both normal and edge cases) (#830)

### To-Do

- [ ] Clean up commit history
- [ ] Add or update documentation related to these changes
- [ ] Add entry to the [release notes](https://github.com/libp2p/py-libp2p/blob/main/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by bomanaps** [comment]
@acul71 @seetadev

**Comment by acul71** [comment]
@bomanaps 
I can see some lint errors, please fix them so we can see all test passes.
before submitting 
- Check if are in sync with `py-libp2p` main branch
run 
- `make pr` and `make docs` (or `make linux-docs` )
To see everything is ok.
I was wandering did you commit skipping checks ? `git commit --no-verify`
Anyway just tag me if you need help with lint errors or anything else...
Reply to comment https://github.com/libp2p/py-libp2p/issues/825#issuecomment-3179145561 also otherwise we could break code base when we merge.

**Comment by bomanaps** [comment]
> @bomanaps I can see some lint errors, please fix them so we can see all test passes. before submitting
> 
>     * Check if are in sync with `py-libp2p` main branch
>       run
> 
>     * `make pr` and `make docs` (or `make linux-docs` )
>       To see everything is ok.
>       I was wandering did you commit skipping checks ? `git commit --no-verify`
>       Anyway just tag me if you need help with lint errors or anything else...
>       Reply to comment [Test Notify Issues #825 (comment)](https://github.com/libp2p/py-libp2p/issues/825#issuecomment-3179145561) also otherwise we could break code base when we merge.

I have fixed the linting issue and all the tests are passing now plus other comment you made

File: libp2p/network/connection/swarm_connection.py [hunk 1]
```diff
@@ -43,6 +43,18 @@ def __init__(
...
+        # Provide back-references/hooks expected by NetStream
+        try:
+            setattr(self.muxed_conn, "swarm", self.swarm)
+
+            # NetStream expects an awaitable remove_stream hook
+            async def _remove_stream_hook(stream: NetStream) -> None:
+                self.remove_stream(stream)
+
+            setattr(self.muxed_conn, "remove_stream", _remove_stream_hook)
+        except Exception:
```
Comment (acul71) [inline libp2p/network/connection/swarm_connection.py]: Can you add a logging message that shows the exception error

**Comment by acul71** [comment]
Well done @bomanaps 

The implementation is solid, well-tested, and follows the established patterns. The code is production-ready for the `closed_stream` functionality
Just answer the `Exception` message
Last thing: please add a newsfragment for the mod. See here: https://github.com/libp2p/py-libp2p/blob/main/newsfragments/README.md (it's a brief desc of the mod)

**Comment by acul71** [comment]
@seetadev 
I've reviewed this PR, look solid, is ready to be merged
